packages/compass-sidebar/src/modules/connection-info.ts

```diff
@@ -25,11 +40,23 @@ export const INITIAL_STATE = {
   connectionStorage: ConnectionStorage,
 };
 
-async function saveConnectionInfo(connectionInfo, connectionStorage) {
+export interface ConnectionInfoState {
+  connectionInfo: Omit<ConnectionInfo, 'id'> & Partial<ConnectionInfo>;
+  connectionStorage: typeof ConnectionStorage;
+}
+
+export type ConnectionInfoAction =
+  | ChangeConnectionInfoAction
+  | SaveConnectionInfoAction;
+
+async function saveConnectionInfo(
+  connectionInfo: ConnectionInfo,
+  connectionStorage: typeof ConnectionStorage
+) {
   try {
-    await connectionStorage.save(connectionInfo);
+    await connectionStorage.save({ connectionInfo });
```

Comment:
^ This is what was preventing connections from being saved

Reply:
Like completely?

Reply:
Yeah, we don’t have an e2e test for this